app/indexers/work_indexer.rb

```diff
@@ -61,6 +61,19 @@ class WorkIndexer < Kithe::Indexer
       acc.concat(DateIndexHelper.new(record).expanded_years)
     end
 
+    # Note that a Work can be associated with two consecutive containers,
+    # in which case e.g. the box number can be something like "34-35".
+    # Searches for box 35 should match any work that is associated with box 35.
+    to_field "box_isim" do |record, acc|
+      val = record&.physical_container&.box
+      acc.concat val.scan(/\d+/) if val.present?
+    end
+
+    to_field "folder_isim" do |record, acc|
+      val = record&.physical_container&.folder
+      acc.concat val.scan(/\d+/) if val.present?
+    end
+
```

Comment:
`select distinct json_attributes -> 'physical_container' -> 'folder' from kithe_models` and `select distinct json_attributes -> 'physical_container' -> 'box' from kithe_models` will yield dozens of them.

So single-valued string, `ssi`, is appropriate, yes.

About normalization -- we can certainly define an all-lowercase normalized field, but it's also easy to just make the search case-insensitive.

Reply:
How is it easy to make the search case insensitive?

Ah, I guess you'd have to down-case at indexing time *in ruby code* too?

I think it's probably better to do all normalization in solr code instead of ruby, where it can be absolutely enforced. Plus the possible unicode normalization issue. It should not be at all hard to define a custom field in Solr that is like a string but with some normalization -- I think you want the "keyword tokenizer" to not actually tokenize at all.

Reply:
After more conversation on Slack with Jonathan and the archivists, and some examination of the extant metadata, we're using text, not string.